src/include/txn_inline.h

```diff
@@ -811,6 +821,20 @@ __wt_upd_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value, u_int modify_type
     }
     upd->type = (uint8_t)modify_type;
 
+#ifdef HAVE_DIAGNOSTIC
+    if (F_ISSET(session->dhandle, WT_DHANDLE_VERB_TS_WRITE)) {
```

Comment:
When we start using these flags for the assertion checking, then this condition needs to be changed.